File: api/golang/engine/lib/kurtosis_context/service_logs_stream_content.go
package kurtosis_context

import "github.com/kurtosis-tech/kurtosis/api/golang/core/lib/services"

// This struct wrap the information returned by the user service logs GRPC stream
type serviceLogsStreamContent struct {
	serviceLogsByServiceUuids map[services.ServiceUUID][]*ServiceLog
	notFoundServiceUuids      map[services.ServiceUUID]bool
}

func newServiceLogsStreamContent(
	serviceLogsByServiceUuids map[services.ServiceUUID][]*ServiceLog,
	notFoundServiceUuids map[services.ServiceUUID]bool,
) *serviceLogsStreamContent {
	return &serviceLogsStreamContent{
		serviceLogsByServiceUuids: serviceLogsByServiceUuids,
		notFoundServiceUuids:      notFoundServiceUuids,
	}
}

func (streamContent *serviceLogsStreamContent) GetServiceLogsByServiceUuids() map[services.ServiceUUID][]*ServiceLog {
	return streamContent.serviceLogsByServiceUuids
}

func (streamContent *serviceLogsStreamContent) GetNotFoundServiceUuids() map[services.ServiceUUID]bool {
	return streamContent.notFoundServiceUuids
}

File: api/typescript/src/engine/lib/kurtosis_context/service_logs_stream_content.ts
import {ServiceUUID} from "../../../core/lib/services/service";
import {ServiceLog} from "./service_log";

//This struct wrap the information returned by the user service logs GRPC stream
export class ServiceLogsStreamContent {
    private readonly serviceLogsByServiceUuids: Map<ServiceUUID, Array<ServiceLog>>;
    private readonly notFoundServiceUuids: Set<ServiceUUID>;

    constructor(
        serviceLogsByServiceUuids: Map<ServiceUUID, Array<ServiceLog>>,
        notFoundServiceUuids: Set<ServiceUUID>,
    ) {
        this.serviceLogsByServiceUuids = serviceLogsByServiceUuids;
        this.notFoundServiceUuids = notFoundServiceUuids;
    }

    // Docs available at https://docs.kurtosis.com/sdk#getservicelogsbyserviceuuids----mapserviceuuid-arrayservicelog-servicelogsbyserviceuuids
    public getServiceLogsByServiceUuids(): Map<ServiceUUID, Array<ServiceLog>> {
        return this.serviceLogsByServiceUuids;
    }

    // Docs available at https://docs.kurtosis.com/sdk#getnotfoundserviceuuids---setserviceuuid-notfoundserviceuuids
    public getNotFoundServiceUuids(): Set<ServiceUUID> {
        return this.notFoundServiceUuids;
    }
}
